lms/services/grouping.py

```diff
@@ -97,6 +100,23 @@ def upsert_canvas_group(  # pylint: disable=too-many-arguments
             )
         )
 
+    def upsert_grouping_memberships(self, user: User, groups: List[Grouping]):
+        memberships = []
+        for group in groups:
+            if membership := (
+                self._db.query(GroupingMembership)
+                .filter_by(grouping=group, user=user)
+                .one_or_none()
+            ):
+                memberships.append(membership)
```

Comment:
I'm not sure this will update the "updated" field which would be interesting to do.

Reply:
Hmm. Yeah we might have to add an extra line to update `updated` manually